ws/v2.md

```diff
@@ -110,7 +110,8 @@ Updates can be sent in bulk since the payload of the message specifies a collect
               "makerAssetData": "0xf47261b04c32345ced77393b3530b1eed0f346429d",
               "takerAssetData": "0x0257179264389b814a946f3e92105513705ca6b990",
               "exchangeAddress": "0x12459c951127e0c374ff9105dda097662a027093",
-              "signature": "0x012761a3ed31b43c8780e905a260a35faefcc527be7516aa11c0256729b5b351bc33"
+              "signature": "0x012761a3ed31b43c8780e905a260a35faefcc527be7516aa11c0256729b5b351bc33",
+              "timestamp": 1537375071174
```

Comment:
The timestamp field can't go in here, since this is a 0x signed order instance. Everything in the `order` object should not be changing. Would a top-level `timestamp` under `requestId` be enough?

Reply:
Yes, sorry about that. Top-level would be great! We're going to need something similar for the REST calls as well. The aggregators need a way to be able to sort the updates of an order.

Reply:
Hi Fragosti, if there are two filled events applied in the shared liquidity pool of all relayers, how can we identify which event came first?

For example, if an aggregator scrapes two new filled events of the same order from two different relayers, what is the easiest way to identify the order of these events? They are both valid and to me it would be via a timestamp. Relying on metadata (remainingTakerAssetAmount) that is not enforced leaves the order of events prone to holes.

As an aggregator we want to enrich data as much as possible for all relayers, and enforcing the order of events via timestamps would be very useful. If there’s another way, let me know but the chaining of events is important to us.

Reply:
@wildnothing the main source of truth for `events` is the blockchain. Imho an aggregator should not have to rely on SRA to get up-to-date stateful information, it should only use it to get existing and new orders from a relayers orderbook. It should then use something like [OrderWatcher](https://0xproject.com/docs/order-watcher) to monitor those orders.

I guess I don't see how the ordering of events matters in this scenario. I am also unsure what you mean by `events` coming from relayers. There are definitely update events coming in the websocket API, but I think in a production system you should only rely on 'new order added' type events, and nothing that has to do with filling. You should be watching the blockchain yourself for that information.

Reply:
The point of the aggregators is to give a real-time snapshot of the liquidity pool to the relayers and anybody else that wants to listen in. The sooner we can transmit state of an order the better for the protocol and the community. It makes sense to me to bind the timestamp to any successful order update and transmit it in a similar manner as the WS v2 spec. Spoke to @fabioberger about this and he said target the aggregation of relayers first before watching the blockchain so I would be interested in what he has to say about this.

The timestamp issue comes back to race conditions. We want the fastest, simplest way to know which order updates came before others so that we can relay this information back to the liquidity pool correctly, as soon as possible.

Needing to watch the blockchain for updates of orders seems like unnecessary overhead when we could just enforce it in the ws + rest response payload.